### PR TITLE
VIH-9001 Restrict Joh Consultation Room

### PR DESCRIPTION
**Pull request by abbasiTahir**
**Before creating a pull request make sure that:**

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/VIH-9001

### Change description ###
Added a Redis cache class to implement the locking .   
The new change blocks the creation of new room when two parallel request are sent . It will delay the second request to prevent the creation of a new room.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: VideoWeb/VideoWeb/Controllers/ConsultationsController.cs [hunk 1]
```diff
@@ -222,6 +226,23 @@ public async Task<IActionResult> StartConsultationAsync(StartPrivateConsultation
...
                     }
+
+                    var johConsultationRoomLockedStatusKeyName = $"johConsultationRoomLockedStatus_{conference.Id}";
+                    var isLocked =
+                        await _distributedJohConsultationRoomLockCache.IsJOHRoomLocked(johConsultationRoomLockedStatusKeyName);
+
+                    if (isLocked)
+                    {
+                        Thread.Sleep(30000); //this is really not ideal, but I cant think of any other options, this should be done from kinly
+                        // await _distributedJohConsultationRoomLockCache.UpdateJohConsultationRoomLockStatus(false,
```
Comment (oliver-scott) [inline VideoWeb/VideoWeb/Controllers/ConsultationsController.cs:237]: can we remove this commented out code?

**Review by oliver-scott — COMMENTED** [review]
This PR is referencing VIH-9067 (in the title and the description) - can you change it to reference VIH-9001 instead (the original bug).

Otherwise looks good to me

File: VideoWeb/VideoWeb.Common/Caching/DistributedJOHConsultationRoomLockCache.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+namespace VideoWeb.Common.Caching
+{
+    public class DistributedJOHConsultationRoomLockCache : RedisCacheBase<string , bool>,IDistributedJOHConsultationRoomLockCache
+    {
+        public DistributedJOHConsultationRoomLockCache(IDistributedCache distributedCache) : base(distributedCache)
+        {
+            CacheEntryOptions = new DistributedCacheEntryOptions
+            {
+                SlidingExpiration = TimeSpan.FromMinutes(5) //reduce this to few seconds?
```
Comment (will-craig) [inline VideoWeb/VideoWeb.Common/Caching/DistributedJOHConsultationRoomLockCache.cs:13]: I dont think the lock needs to last 5 minutes. It should only be very temporary, like 5 seconds then expire

File: VideoWeb/VideoWeb/Controllers/ConsultationsController.cs [hunk 3]
```diff
@@ -222,6 +226,23 @@ public async Task<IActionResult> StartConsultationAsync(StartPrivateConsultation
...
                         return Forbid();
                     }
+
+                    var johConsultationRoomLockedStatusKeyName = $"johConsultationRoomLockedStatus_{conference.Id}";
+                    var isLocked =
+                        await _distributedJohConsultationRoomLockCache.IsJOHRoomLocked(johConsultationRoomLockedStatusKeyName);
+
+                    if (isLocked)
+                    {
+                        Thread.Sleep(30000); //this is really not ideal, but I cant think of any other options, this should be done from kinly
```
Comment (will-craig) [inline VideoWeb/VideoWeb/Controllers/ConsultationsController.cs:236]: This is too long to pause the thread in my opinion, 30 seconds to keep the user hanging is not a good user experiece. If the redis cache lock is expires after 5 seconds (see above comment) then it should be no more than that long

File: VideoWeb/VideoWeb/Controllers/ConsultationsController.cs (hunk 1 above)
Comment (abbasiTahir) [reply VideoWeb/VideoWeb/Controllers/ConsultationsController.cs:237]: Removed .

File: VideoWeb/VideoWeb/Controllers/ConsultationsController.cs (hunk 3 above)
Comment (abbasiTahir) [reply VideoWeb/VideoWeb/Controllers/ConsultationsController.cs:236]: wait _consultationService.GetAvailableConsultationRoomAsync(request.ConferenceId, request.RoomType.MapToDomainEnum()).  Not an video-api n endpoint . I will reduce the thread delay however whilst demoing to komal i did not noticed any significant delay creation of room .  I will leave a 3 seconds delay .

**Review by oliver-scott — APPROVED** [review]

**Pull request merged**